### PR TITLE
One-to-many recursion translation rule + latexification

### LINKED ISSUE #22: Justify reasoning in the one-to-many recursion rule
The translation rule does not explain _why_ the value computed by teh alternate operator is the same as the one computed by the original operator, it merely defines the syntax.

### PR DESCRIPTION
**Pull request by Kukovec**
Adds a special translation rule, for the case where a recursive operator computes multiple values recursively (instead of a single recursive call).

closes #22

### CODE REVIEW

**Review by konnov — COMMENTED** [review]
I have to read this carefully. This reminds me of recursive functions (not operators) in TLA+ a bit

**Comment by konnov** [comment]
@Kukovec is it still in the draft stage?

**Comment by Kukovec** [comment]
Technically, the rule itself is complete, what's missing is 
https://github.com/freespek/ssf-mc/blob/e64112b08780409153e2644609b6123e3ac2e03a/Translation.md?plain=1#L558

**Review by konnov — COMMENTED** [review]
Could you add a makefile, so we don't have to re-run `pdflatex` multiple times by hand?

**Comment by Kukovec** [comment]
Note that this makes `Translation.md` deprecated, do we want to just delete it as part of this pr?

**Comment by konnov** [comment]
> Note that this makes `Translation.md` deprecated, do we want to just delete it as part of this pr?

Yes, if it's not needed anymore, let's remove it

**Comment by thpani** [comment]
Could you please change those commit messages to something meaningful?

**Comment by banhday** [comment]
@konnov : I will try to have a look at it by this Wednesday.

**Comment by banhday** [comment]
@Kukovec : I have added a few questions in the *.tex files.

File: reports/appendix.tex [hunk 1]
```diff
@@ -39,7 +39,9 @@ \subsection{Proofs}
 \end{lemma}
 \begin{proof}
 Let $g \coloneqq \mop(f)$. To prove the first part, we have two options to consider, based on the definition of $\op$. If $\forall x \in D_f \ .\ P(x)$, then $D_f = D_g$.
-Otherwise, $g = \mop(f,\mop(\bb(f)))$. By definition, $D_{\mapg(f,m)} = D_f \cup D_m$, so it follows that $D_f \subseteq D_{\mapg(f,m)}$, for any $m$. 
+Otherwise, $g = \mop(f,\mop(\bb(f)))$ 
+\tth{Should it have been $g = \mapg(f,\mop(\bb(f)))$ ? }
```
Comment (Kukovec) [inline reports/appendix.tex:43]: Yes, that's a typo

File: reports/appendix.tex [hunk 2]
```diff
@@ -61,7 +63,10 @@ \subsection{Proofs}
 % we do not need to consider the second branch of the definition to compute $\op(x)$. Thus, we do not need to recursively compute $\op(v)$ for any $v$, and $h(V(x))$ is empty.
 % \end{proof}
 
-We define an auxiliary function $\alpha$ that assigns every function of the type $a \to Set(a)$ a value in $\nat \cup \{-\infty\}$, defined as:
+We define an auxiliary function $\alpha$ that assigns every function of the type $a \to Set(a)$ a value in $\nat \cup \{-\infty\}$, defined as
+\tth{Does sup refer to supremum?}
```
Comment (Kukovec) [inline reports/appendix.tex:67]: yes

File: reports/appendix.tex [hunk 3]
```diff
@@ -61,7 +63,10 @@ \subsection{Proofs}
...
+\tth{Under what condition $\alpha(f) = -\infty$? Does there exist any case where $\alpha(f) = \infty$?}
```
Comment (Kukovec) [inline reports/appendix.tex:68]: By definition, the supremum is the _lowest upper bound_. For the case where a set is empty, every value `v` is an upper bound by the vacuously true formula
```
\A x \in {}: v >= x 
```
so for the empty set the _lowest_ upper bound does not exist (cannonically denoted `sup {} = -inf`). Conversely, if no upper bound existed at all (i.e. if the set contained arbitrarily large elements) we would say `sup S = inf`. Observe that this is only possible for infinite sets, which we are not dealing with, so `alpha(f) = inf` is impossible in our setting.

File: reports/appendix.tex [hunk 4]
```diff
@@ -71,6 +76,7 @@ \subsection{Proofs}
 \[
 \alpha(f) \ge 0 \Rightarrow \alpha(b(f)) < \alpha(f)
 \]
+\tth{What is $b(f)$?}
```
Comment (Kukovec) [inline reports/appendix.tex:79]: Typo, should be `\bb`

File: reports/meta.tex [hunk 5]
```diff
@@ -26,7 +26,9 @@ \subsubsection{Bounded recursion rule}
   \item A general case operator $G$, which invokes a recursive computation of $R$ over a modified parameter given by the operator $\bb$.
 \end{itemize}
 %
-The following needs to hold true, to ensure recursion termination: for every $x\colon a$, there exists a finite sequence $x = v_1, \dots, v_n$, such that
+The following needs to hold true, to ensure recursion termination: for every $x\colon a$, there exists a finite sequence $x = v_1, \dots, v_n$
+\tth{Should we use the same notation $\tup{v_n, \ldots, x}$ for sequences as on Page 10.}
```
Comment (Kukovec) [inline reports/meta.tex:30]: We could, but I don't like propagating TLA notation into proofs if it can be avoided.

File: reports/meta.tex [hunk 6]
```diff
@@ -48,7 +50,10 @@ \subsubsection{Bounded recursion rule}
   IN ApaFoldSeqLeft( step, <<x>>, MkSeq(N, LAMBDA i: i) )
 \end{lstlisting}
 %
-We can see that $\Chain(x,N)$ returns the sequence $\tup{v_n, ..., x}$ if $N$ is sufficiently large. We can verify whether or not that is the case, by evaluating $P(\Chain(x, N)[1])$. If it does not hold, the $N$ chosen is not large enough, and needs to be increased. Using $\Chain$ we can define a fold-based non-recursive operator $\op^*$, such that $\op^*(x) = \op(x)$ under the above assumptions:
+We can see that $\Chain(x,N)$
+\tth{Some lines in \tlap{} are too long.} returns the sequence $\tup{v_n, ..., x}$ if $N$ is sufficiently large
+\tth{Can you please provide an intuition of functions \texttt{ApaFoldSeqLeft} and \texttt{MkSeq}>}.
```
Comment (Kukovec) [inline reports/meta.tex:55]: Sure

File: reports/meta.tex [hunk 7]
```diff
@@ -167,7 +172,9 @@ \subsubsection{ Mutual recursion cycles}
       )
     )
 \end{lstlisting}
-for which $\op(x) = \op_1(x)$ for all $x$, and $\op$ terminates iff $\op_1$ terminates.
+for which $\op(x) = \op_1(x)$ for all $x$, and $\op$ terminates iff $\op_1$ terminates
+\tth{Should we rename the funcdtion as \texttt{R(\_)} is already used on Pages 9 and 10?}
```
Comment (Kukovec) [inline reports/meta.tex:176]: The point of this section is to show that we can define an operator to which the rewriting rules of the previous section could apply. The previous section does not define "R" but rather express the shape assumptions on it.

File: reports/appendix.tex (hunk 1 above)
Comment (konnov) [reply reports/appendix.tex]: Fixed

**Review by konnov — APPROVED** [review]
Let's merge this document for the interim reporting purposes. We can further polish in the final report.

**Comment by konnov** [comment]
I have merged this pull request. @Kukovec, please address the remaining comments in the follow-up PR. @banhday, many thanks for proof-reading!

**Pull request merged**